content/articles/feedforward-and-recurrent-neural-networks-python-implimentation/index.md

```diff
@@ -0,0 +1,281 @@
+
+```python
+a1 = num.random.randn(inputlayer_dimensionality, hiddenlayer_dimensionality)
+c1 = num.zeros((1, hiddenlayer_dimensionality))
+
+a2= num.random.randn(hiddenlayer_dimensionality, hiddenlayer_dimensionality)
+c2 = num.zeros((1, hiddenlayer_dimensionality))
+
+a3= num.random.randn(hiddenlayer_dimensionality, outputlayer_dimensionality)
+c3 = num.zeros((1, outputlayer_dimensionality))
+```
+
+#### Step four: Foward propagation of the input signal 
+To reach the output layer, the propagation will take place over several layers.
+
+```python
+d1 = X.dot(a1) + c1
+q1 = num.tanh(z1)
+
+d2 = q1.dot(a2) + c2
+q2 = num.tanh(z2)
+
+d3 = q2.dot(a3) + c3
+
+probs = num.exp(d3) / num.sum(num.exp(d3), axis=1, keepdims=True)
+```
+
+A linear neural network is one that does not have any activation functions in any of its layers. Non-linear neural networks are those that have action functions such as the relu, sigmoid, or tanh in any of their layers, or even in multiple layers.
+
+In most cases, joining two different curves results in a more complicated shape. In addition to introducing hyper-curves like hyperplanes and non-linear objects, these activation functions introduce non-linearity at every layer. When we apply nonlinear functions to nonlinear objects, we're creating a nonlinear entity out of a nonlinear object.
+
+A single linear regression model can simply replace a neural network in the absence of an activating function.
+
+$y = ax + d$
+
+$Q= ky + t $
+
+$=k(ax+d) + t $
+
+$=kax + kd + t$
+
+$ =(ka)x + (kc+t)$
+
+### Applications of Feedforward neural network
+1. An illustrious network of genetic regulation and feedforward has been found to be a feedforward system for the detection of non-temporary atmospheric modifications.
+2. In the field of automation and machine management, feedforward control may be a discipline.
+3. Derivative parallel feedforward compensation can be used to transform an open-loop transfer system into one that operates at its minimum.
+
+### Recurrent neural network(RNN)
+One of the most frequent types of artificial neural networks, it is called a recurrent neural network. As a tool for automatic voice recognition and machine translation (RNN). It is possible to forecast the most likely future situation utilizing patterns in sequential data by employing recurrent neural networks.
+### Recurrent neural network gradient problems 
+Recurrent neural networks, as strong as they are, are vulnerable to gradient-related training issues. The $n$ derivatives of a network with $n$ hidden layers will be multiplied together. When these derivatives are significant, the gradient increases exponentially as it propagates backwards until it bursts. This is known as the `exploding gradient` problem. A problem known as the `vanishing gradient` problem occurs when the gradient propagates yet the derivatives are so small that the gradient finally vanishes.
+
```

Comment:
Why do you specifically talk about the gradient problem of RNN? Is it not an introductory article?